### PR TITLE
Fix service catalog upgrade from 3.6

### PR DESCRIPTION
**Pull request by jpeeler**
This enables the aggregator (for all upgrades), which is required for
service catalog to function as expected.

---

Still working one last remaining issue.

### CODE REVIEW

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -1927,7 +1927,7 @@ def get_defaults(self, roles, deployment_type, deployment_subtype):
                                       console_use_ssl=True,
                                       console_path='/console',
                                       console_port='8443', etcd_use_ssl=True,
-                                      etcd_hosts='', etcd_port='4001',
+                                      etcd_hosts='', etcd_port='2379',
```
Comment (liggitt) [inline roles/openshift_facts/library/openshift_facts.py]: is this change intentional?

Comment (jpeeler) [reply roles/openshift_facts/library/openshift_facts.py]: Yes, it is. I think it's an outdated default and would rather just use it than continue to override it.

**Comment by sdodson** [comment]
Looks like this would fix the bug by ensuring that the api is wired in during the upgrade. However I think we should still include that set of tasks in the service_catalog playbook, they may not re-run the upgrade playbook before attempting to re-run the service catalog playbook again so we may as well include it there too.

File: roles/openshift_facts/library/openshift_facts.py (hunk 1 above)
Comment (sdodson) [reply roles/openshift_facts/library/openshift_facts.py]: no, everywhere overrides it as far as i know.

File: playbooks/common/openshift-master/tasks/wire_aggregator.yml [hunk 2]
```diff
@@ -142,10 +142,11 @@
...
 
-- name: Setup extension file for service console UI
-  template:
-    src: ../templates/openshift-ansible-catalog-console.js
-    dest: /etc/origin/master/openshift-ansible-catalog-console.js
+# setup extension file for service console UI
+- lineinfile:
+    dest=/etc/origin/master/openshift-ansible-catalog-console.js
+    line="window.OPENSHIFT_CONSTANTS.TEMPLATE_SERVICE_BROKER_ENABLED={{ 'true' if (template_service_broker_install | default(True)) else 'false' }}"
+    create=yes
```
[the remark this answers is not in the document]
Comment (sdodson) [reply playbooks/common/openshift-master/tasks/wire_aggregator.yml]: I just mean yaml `key: value` versus json `key=value` for dest, line, create. everything else is fine with this task.

**Comment by jpeeler** [comment]
I believe all the latest CI failures are flakes/bugs in current CI env. And although it was requested for service catalog playbooks to be updated to also include turning on the aggregation, I'm requesting that be done as a separate follow up to this work.

**Comment by sdodson** [comment]
@jpeeler That's fine with me thanks for digging into this one. Reviewing the test logs those are common flakes so going ahead with the merge.

**Comment by wujm2017** [comment]
me too!!! I check the configuration is correct.Always make the mistake.

TASK [openshift_service_catalog : wait for api server to be ready] ****************************************************************************************************
FAILED - RETRYING: wait for api server to be ready (120 retries left)
